definitions/types/aws-arn.json

```diff
@@ -7,7 +7,7 @@
     "arn:aws:rds::ACCOUNT_NUMBER:db/prod",
     "arn:aws:ec2::ACCOUNT_NUMBER:vpc/vpc-foo"
   ],
-  "pattern": "^arn:aws:[a-zA-Z0-9._-]*:[a-zA-Z0-9._-]*:(?:[0-9]{12})?:[a-zA-Z0-9._-]+(?:[/:][a-zA-Z0-9/._-]+)?(?:/[0-9-T:.]+)?$",
+  "pattern": "^arn:aws:[a-zA-Z0-9._-]*:[a-zA-Z0-9._-]*:(?:[0-9]{12})?:[a-zA-Z0-9._-]+(?:[/:][a-zA-Z0-9:/._-]+)$",
```

Comment:
We have been making more specialized arn types given they can have arbitrarily long tails. We should consider that here.

Reply:
The reason for the change is that the current regex pattern doesn't support the managed RabbitMQ ARN, so I made it slightly more permissive to support it. I can instead update the final block to support the RabbitMQ ARN, but it does beg the question of how specific we want to get w/ the types/patterns. Technically we could have a custom ARN pattern for every AWS artifact, but that defeats the purpose of many of `types`, since the same argument could be made for `gcp-grn` and `azure-ari`.

My understanding was that the main purpose of an `aws-arn` type would simply be to make sure the field "looked" like an ARN (instead of just a resource name). Let's discuss more tomorrow.